pkg/api/defaults-kubelet.go

```diff
@@ -128,17 +128,25 @@ func (cs *ContainerService) setKubeletConfig() {
 		if profile.KubernetesConfig == nil {
 			profile.KubernetesConfig = &KubernetesConfig{}
 			profile.KubernetesConfig.KubeletConfig = make(map[string]string)
-			if profile.OSType == "Windows" {
-				for key, val := range staticWindowsKubeletConfig {
-					profile.KubernetesConfig.KubeletConfig[key] = val
-				}
+		}
+
+		if profile.OSType == "Windows" {
+			for key, val := range staticWindowsKubeletConfig {
+				profile.KubernetesConfig.KubeletConfig[key] = val
 			}
 		}
+
 		setMissingKubeletValues(profile.KubernetesConfig, o.KubernetesConfig.KubeletConfig)
 
 		if profile.OSType == "Windows" {
 			// Remove Linux-specific values
 			delete(profile.KubernetesConfig.KubeletConfig, "--pod-manifest-path")
+
+			if !to.Bool(o.KubernetesConfig.EnableSecureKubelet) {
```

Comment:
let's add this to existing unit tests

Reply:
Since @xizhamsft found and fixed a bug in our implementation, I can offer unit tests :)

@xizhamsft could you check over my logical tweaks and confirm that things are as they should be? (Also validate that the unit tests express functional intent)

Sorry, this code area is a bit hairy!

Reply:
@jackfrancis , could you push your change again? I did notice you made some changes too and did a force push. Now your changes are lost

Reply:
@jackfrancis , LGTM